ambry-api/src/main/java/com.github.ambry/config/NettyConfig.java

```diff
@@ -73,6 +73,16 @@
   @Default("8192")
   public final int nettyServerMaxChunkSize;
 
+  /**
+   * The threshold of the size of buffered data at which reading from a client channel will be suspended. If the size
+   * drops below the threshold, reading will be resumed. This value is respected on a per-request basis.
+   * Note that the actual amount of data buffered will be >= this number.
```

Comment:
@pnarayanan "may be" sounds better will change
@xiahome I think this is fine - and since this is a configuration, can always be tuned.